### PR TITLE
feat(connector): add dummy connector template code

### PR DESCRIPTION
**Pull request by ThisIsMani**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Add template code for dummy connector and feature flags to enable and disable it.

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

#726 

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by SanchithHegde** [comment]
When you're opening PRs that fix an open issue, please link the issue to your PR by using [one of the keywords recognized by GitHub](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword). It makes tracking issues and PRs easier for us. For this PR, I've done it for you, please try to follow it for further PRs you open.

**Review by SanchithHegde — APPROVED** [review]

**Review by NishantJoshi00 — DISMISSED** [review]
Please raise the sub-sequent PRs that add code for this template and link this PR to them

**Comment by ashokkjag** [comment]
Hi @ThisIsMani , Is dummyconnector planned to behind a compile time feature flag? I thought the plan was to also use it as dummy connector for any customer who signs up, to experience the dashboard.

**Comment by ThisIsMani** [comment]
> Hi @ThisIsMani , Is dummyconnector planned to behind a compile time feature flag? I thought the plan was to also use it as dummy connector for any customer who signs up, to experience the dashboard.

@ashokkjag, yes the usecase for the dummyconnector is the same as you said. I wanted to make it as an option to have the dummyconnector to be disabled using a feature flag.

**Review by ashokkjag — APPROVED** [review]

**Review by jarnura — APPROVED** [review]

File: crates/api_models/src/enums.rs [hunk 1]
```diff
@@ -592,6 +592,10 @@ pub enum Connector {
     Cybersource,
     #[default]
     Dummy,
```
[the remark this answers is not in the document]
Comment (ThisIsMani) [reply crates/api_models/src/enums.rs:594]: Yes, it will become default once the Dummy Connector implementation is done.

**Pull request merged**